docs/layer_reference/norm.rst

```diff
@@ -32,3 +32,10 @@ Dropout Layer
 .. autoclass:: returnn.tf.layers.basic.DropoutLayer
     :members:
     :undoc-members:
+
+Tikhonov Regularization Layer
```

Comment:
I'm really not sure whether it is a good idea to document layers which are probably deprecated and should not be used (or are not really used). This just bloats the documentation. And even worse, it could lead people to think this is well supported, and then we later deprecate it or remove it, or they stumble upon bugs.

(If you are not sure whether a layer is used or deprecated, just ask. But if you have never used it yourself, or maybe don't even know what this is for, this is already a good sign that it is not used.)

Reply:
Also, why is this in `norm.rst`? This is not about normalization.

Reply:
This is what I wanted do discuss in last weeks meeting, maybe we can do that today. I already left out many layers where I thought that they are deprecated or unnecessary, but for some I were not sure so I included them for review.

Reply:
The section is also about any kind of regularization, so it is correct here. I can change the file name if you want, but I did not consider it necessary.

 So you wrote the Tikhonov regularization, do you think this should not be used?

Reply:
> The section is also about any kind of regularization, so it is correct here.

I don't understand. The filename suggests this is about normalization, not regularization. What is it? Both? Why both? Normalization and regularization are very orthogonal concepts. I would not mix them.

> I can change the file name if you want, but I did not consider it necessary.

If this is about regularization, then the filename should be `regularization.rst` or so.

If this is about both normalization and regularization, the filename should be `normalization_and_regularization.rst`, but actually, it should be split, as these are independent concepts.

> So you wrote the Tikhonov regularization, do you think this should not be used?

It was an idea by Ney. It was used in exactly one experiment. Maybe I should not have pushed it directly to RETURNN. Originally I thought it would be more useful in general, but then it was never used again.